packages/picasso.js/src/core/component/component-factory.js

```diff
@@ -475,6 +477,17 @@ function componentFactory(definition, context = {}) {
       }
     });
 
+    if (
+      settings.rendererSettings &&
+      typeof settings.rendererSettings.transform === 'function' &&
+      settings.rendererSettings.transform()
+    ) {
+      rend.render();
+      return;
+    }
+
+    const nodes = (brushArgs.nodes = render.call(definitionContext, ...getRenderArgs()));
+
```

Comment:
You are right about that. Good catch again. I moved the code back up.

Reply:
I don't think so... But actually I'm a little unsure about that, I think we might do some tuning on that in the future if we see the need for it. I just added this little note in the description of the transform feature:

"!Important: When a transform is applied to a component, the underlaying node representations are not updated with the new positions/sizes, which\ncan cause problems for operations that relies on the positioning of the shapes/nodes (such as tooltips, selections etc). An extra chart update\nwithout a transform is therefore needed to make sure the node position information is in sync with the visual representation again."

This means it's probably not suitable to use transforms as the same time as brushing (since the position of the shapes are not in sync with the visual representation).... So I think it's fine like this for now at least